### PR TITLE
Add initial integration tests (email+extension)

### PR DESCRIPTION
**Pull request by shakyShane**
**Reviewer:** @jonathanKingston 
**Asana:** https://app.asana.com/0/0/1201860080529499/f

## Description

We need to start adding integration tests - this PR adds what I consider to be the bare-minimum (it just checks the extension messaging works for email auto fill)

It's set to run in CI also and once merged we can look to mimic the webkit message handlers which will enable us to test a lot of the apple logic too.

### CODE REVIEW

File: integration-test/helpers/harness.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        // const evalString = `
+        //     const detail = ${JSON.stringify(args)}
+        //     const evt = new CustomEvent('content-scope-init-args', { detail })
+        //     document.dispatchEvent(evt)
+        // `
+        // await page.evaluate(evalString)
+        //
+        // // wait until contentScopeFeatures.init(args) has completed
+        // await page.waitForFunction(() => {
+        //     return window.__content_scope_status === 'initialized'
```
Comment (jonathanKingston) [inline integration-test/helpers/harness.js:122]: I think all these comments can go right? just me copying and pasting.

File: src/DeviceInterface/InterfacePrototype.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ const listenForGlobalFormSubmission = require('../Form/listenForFormSubmission')
 const {forms} = require('../scanForInputs')
 
 class InterfacePrototype {
+    mode = 'test';
```
Comment (jonathanKingston) [inline src/DeviceInterface/InterfacePrototype.js:21]: This shouldn't be the default.

**Comment by shakyShane** [comment]
@jonathanKingston this is ready for review again now, I've moved some of the asset copying into a script to make the variable replacements easier

**Comment by jonathanKingston** [comment]
LGTM thanks for finishing off the PR :)
